File: hyper_bot.py
import time
import requests
import logging
import sys

# Hyperbolic API configuration
HYPERBOLIC_API_URL = "https://api.hyperbolic.xyz/v1/chat/completions"
HYPERBOLIC_API_KEY = "$API_KEY"  # Replaced by Bash script
MODEL = "meta-llama/Llama-3.3-70B-Instruct"
MAX_TOKENS = 2048
TEMPERATURE = 0.7
TOP_P = 0.9
DELAY_BETWEEN_QUESTIONS = 30  # Delay between questions in seconds

logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

def get_response(question: str) -> str:
    headers = {
        "Content-Type": "application/json",
        "Authorization": f"Bearer {HYPERBOLIC_API_KEY}"
    }
    data = {
        "messages": [{"role": "user", "content": question}],
        "model": MODEL,
        "max_tokens": MAX_TOKENS,
        "temperature": TEMPERATURE,
        "top_p": TOP_P
    }
    response = requests.post(HYPERBOLIC_API_URL, headers=headers, json=data, timeout=30)
    response.raise_for_status()
    json_response = response.json()
    return json_response.get("choices", [{}])[0].get("message", {}).get("content", "No answer")

def main():
    # Check if filename is provided as argument
    if len(sys.argv) < 2:
        logger.error("No questions file provided. Usage: python hyper_bot.py <questions_file>")
        sys.exit(1)
    
    questions_file = sys.argv[1]
    
    # Reading questions from the provided file
    try:
        with open(questions_file, "r", encoding="utf-8") as f:
            questions = [line.strip() for line in f if line.strip()]
    except Exception as e:
        logger.error(f"Error reading the file {questions_file}: {e}")
        sys.exit(1)

    if not questions:
        logger.error(f"The file {questions_file} contains no questions.")
        sys.exit(1)

    index = 0
    while True:
        question = questions[index]
        logger.info(f"Question #{index+1}: {question}")
        try:
            answer = get_response(question)
            logger.info(f"Answer: {answer}")
        except Exception as e:
            logger.error(f"Error retrieving answer for question: {question}\n{e}")
        index = (index + 1) % len(questions)
        time.sleep(DELAY_BETWEEN_QUESTIONS)

if __name__ == "__main__":
    main()

File: hyper_bot.js
const fs = require('fs');
const axios = require('axios');

// Hyperbolic API configuration
const HYPERBOLIC_API_URL = "https://api.hyperbolic.xyz/v1/chat/completions";
const HYPERBOLIC_API_KEY = "$API_KEY"; // Replaced by Bash script
const MODEL = "meta-llama/Llama-3.3-70B-Instruct";
const MAX_TOKENS = 2048;
const TEMPERATURE = 0.7;
const TOP_P = 0.9;
const DELAY_BETWEEN_QUESTIONS = 30 * 1000; // Delay in milliseconds

const logger = {
    info: console.log,
    error: console.error
};

async function getResponse(question) {
    const headers = {
        "Content-Type": "application/json",
        "Authorization": `Bearer ${HYPERBOLIC_API_KEY}`
    };
    const data = {
        messages: [{ role: "user", content: question }],
        model: MODEL,
        max_tokens: MAX_TOKENS,
        temperature: TEMPERATURE,
        top_p: TOP_P
    };
    try {
        const response = await axios.post(HYPERBOLIC_API_URL, data, { headers, timeout: 30000 });
        return response.data.choices[0].message.content || "No answer";
    } catch (error) {
        throw error;
    }
}

async function main() {
    let questions;
    try {
        const content = fs.readFileSync("questions.txt", "utf-8");
        questions = content.split('\n').map(line => line.trim()).filter(line => line);
    } catch (error) {
        logger.error(`Error reading the questions.txt file: ${error}`);
        return;
    }

    if (!questions.length) {
        logger.error("The questions.txt file contains no questions.");
        return;
    }

    let index = 0;
    while (true) {
        const question = questions[index];
        logger.info(`Question #${index + 1}: ${question}`);
        try {
            const answer = await getResponse(question);
            logger.info(`Answer: ${answer}`);
        } catch (error) {
            logger.error(`Error retrieving answer for question: ${question}\n${error}`);
        }
        index = (index + 1) % questions.length;
        await new Promise(resolve => setTimeout(resolve, DELAY_BETWEEN_QUESTIONS));
    }
}

main().catch(error => logger.error(`Main error: ${error}`));
